memstore/common/data_value.go

```diff
@@ -32,6 +33,13 @@ var NullDataValue = DataValue{}
 // SizeOfGeoPoint is the size of GeoPointGo in memory
 const SizeOfGeoPoint = unsafe.Sizeof(GeoPointGo{})
 
+// ZeroLengthArrayFlag is the value to represent 0 length array value
+// used in offset part of OffsetLength vector
+// when length part is non-zero value, the offset is real offset in memory
+// when length is 0 and offset is 0, the array value is invalid (by default)
+// when length is 0 and offset is ZeroLengthArrayFlag, this is a 0 length array value
```

Comment:
seems like we just need offset to be nonzero right for this case?

Reply:
offset is not enouth, there is real valid cases when offset is 0

Reply:
no i mean when length is 0 we just need to know whether offset is zero or not to determine invalid or not, right?

Reply:
that is correct